spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+import numpy as np
+import scipy.signal
+import scipy.spatial
+
+from ..postprocessing import compute_correlograms, get_template_extremum_channel
+
+# TODO:
+#   * adaptative window p(aka plage) on CC
+#   * template similarity
+#   * 
+
+def get_potential_auto_merge(waveform_extractor,
+                minimum_spikes=1000, maximum_distance_um=200.,
+                peak_sign="neg",
+                bin_ms=0.25, window_ms=50., corr_thresh=0.3,
+                correlogram_low_pass = 800., adaptative_window_threshold=0.5,
+                debug_folder=None,
+                ):
+    """
+    Algorithm to find and check potential merges between units.
+    
+    This is taken from lussac version 1 done Aurelien Wyngaard.
+    https://github.com/BarbourLab/lussac/blob/main/postprocessing/merge_units.py
+    
+    This check:
+       * correlograms
+       * template_similarity
+       * contamination quality mertics
+    
+    """
+    
+    we = waveform_extractor
+    sorting = we.sorting
+    unit_ids = sorting.unit_ids
+    
+    # STEP 1 : to get fast computation we will not analyse pairs when:
+    #    * not enough spikes for one of theses
+    #    * to far away one from each other
+    n = unit_ids.size
+    pair_mask = np.ones((n, n), dtype='bool')
+    num_spikes = np.array(list(sorting.get_total_num_spikes().values()))
+    to_remove = num_spikes < minimum_spikes
+    pair_mask[to_remove, :] = False
+    pair_mask[:, to_remove] = False
+
+    # STEP 2 : remove contaminated auto corr
+
+    # STEP 3 : unit positions are estimated rougtly with channel
+    chan_loc = we.recording.get_channel_locations()
+    unit_max_chan = get_template_extremum_channel(we, peak_sign=peak_sign, mode="extremum", outputs="index")
+    unit_max_chan = list(unit_max_chan.values())
+    unit_locations = chan_loc[unit_max_chan, :]
+    unit_distances = scipy.spatial.distance.cdist(unit_locations, unit_locations, metric='euclidean')
+    pair_mask = pair_mask & (unit_distances <=maximum_distance_um)
+    
+    print('Will check ', np.sum(pair_mask), 'pairs on ', pair_mask.size)
+    
+    # Step 4 : potential auto merge by correlogram
+    correlograms, bins = compute_correlograms(sorting, window_ms=window_ms, bin_ms=bin_ms, method='numba')
```

Comment:
Shouldn't we give the waveform extractor here rather than the sorting?

Reply:
If we user the waveform extractor we will get the cached version of correlograms with parameters that maybe will not match the wanted params. It is easier to recompute the correlogram.